### PR TITLE
Added istanbul test coverage to project

### PR DESCRIPTION
**Pull request by bendiggle**
- Added istanbul to scripts in package.json
- Added istanbul config yml file
- Added cover to pre-commit (package.json)
- Added coverage to ignore files (docker, eslint)

### CODE REVIEW

**Comment by JoeChapman** [comment]
Don't know why but one of the acceptance tests is failing and breaking the builf

**Comment by bendiggle** [comment]
@JoeChapman Yeah, I've seen that happen before in previous commits. I've just restarted the build. That usually fixes it. I've created a ticket to look in to why the tests fail randomly also.

**Comment by joefitter** [comment]
@JoeChapman @bendiggle could it be a similar issue to the one we were having with hof-example-form and GRO to do with the version of JSDOM? https://github.com/UKHomeOffice/hof-example-form/pull/37/commits/e8a61bd4dee4bf4a3745e04d1807a80a0470d141

**Comment by JoeChapman** [comment]
@bendiggle @joefitter perhaps, @bendiggle can you fold that change into this pr?

File: .istanbul.yml [hunk 1]
```diff
@@ -0,0 +1,21 @@
+instrumentation:
+  root: apps
+  extensions:
+    - .js
+  default-excludes: true
+  include-all-sources: true
+  excludes: ['**/fields/*.js', '**/index.js', '**/steps.js']
```
Comment (JoeChapman) [inline .istanbul.yml]: Do we need to test these files?

Comment (joefitter) [inline .istanbul.yml]: I'm not sure we can, field and steps are just config, we could possibly test index, make sure the middleware is applied correctly etc?

**Comment by JoeChapman** [comment]
LGTM, @joefitter @gxxm

**Comment by joefitter** [comment]
Yeah lets do it

**Comment by JoeChapman** [comment]
go merge

**Comment by joefitter** [comment]
CATCHUP CHAPMAN

**Pull request merged**